LabApiUtilities/src/main/com/microsoft/identity/labapi/utilities/client/LabClient.java

```diff
@@ -198,7 +227,26 @@ public LabGuestAccount loadGuestAccountFromLab(LabQuery labQuery) throws LabApiE
     @Override
     public String getPasswordForGuestUser(LabGuestAccount guestUser) throws LabApiException {
         final String labName = guestUser.getHomeDomain().split("\\.")[0];
-        return getSecret(labName);
+
+        // Adding a second attempt here, api sometimes fails to get the lab secret.
+        try {
+            return getSecret(labName);
+        } catch (LabApiException e){
+            if (e.getErrorCode().equals(LabError.FAILED_TO_GET_SECRET_FROM_LAB)){
```

Comment:
I think we need to find why these things fail sometime. I mean the thread sleep is useful if the API worked but things are still being replicated in the directory, but in this case it seems the API itself is failing and that seems to be an actual bug.

Reply:
So I think doing thread sleeps in this case is not the right approach

Reply:
I think it's possible the issue is connection based, very rarely, some tests fail to perform cloud descovery, so i figured we were failing the secret and other api operations due to flaky internet connection? The 5 second wait is to give the device time to reconnect to a network before attempting another call. I'd be interested in investigating this further but I think it's fine to have this boosting reliability for now and include it in this PR. Thoughts?